File: lib/csharp/se/lth/control/labcomm/LabCommEncoder.cs
namespace se.lth.control.labcomm {

  using System;

  public interface LabCommEncoder {
    
    void register(LabCommDispatcher dispatcher);
    void begin(Type c);
    void end(Type c);
    void encodeBoolean(bool value);
    void encodeByte(byte value);
    void encodeShort(short value);
    void encodeInt(int value);
    void encodeLong(long value);
    void encodeFloat(float value);
    void encodeDouble(double value);
    void encodeString(String value);
    void encodePacked32(Int64 value);
    
  }

}

File: lib/java/se/lth/control/labcomm2006/LabCommEncoder.java
package se.lth.control.labcomm2006;

import java.io.IOException;

public interface LabCommEncoder {

  public void register(LabCommDispatcher dispatcher) throws IOException;
  public void begin(Class<? extends LabCommSample> c) throws IOException;
  public void end(Class<? extends LabCommSample> c) throws IOException;
  public void encodeBoolean(boolean value) throws IOException;
  public void encodeByte(byte value) throws IOException;
  public void encodeShort(short value) throws IOException;
  public void encodeInt(int value) throws IOException;
  public void encodeLong(long value) throws IOException;
  public void encodeFloat(float value) throws IOException;
  public void encodeDouble(double value) throws IOException;
  public void encodeString(String value) throws IOException;
  public void encodePacked32(long value) throws IOException;

}
